lib/config/configuration.go

```diff
@@ -1935,9 +1938,14 @@ func Configure(clf *CommandLineFlags, cfg *service.Config) error {
 
 	// If this process is trying to join a cluster as an application service,
 	// make sure application name and URI are provided.
-	if slices.Contains(splitRoles(clf.Roles), defaults.RoleApp) &&
-		(clf.AppName == "" || clf.AppURI == "") {
-		return trace.BadParameter("application name (--app-name) and URI (--app-uri) flags are both required to join application proxy to the cluster")
+	if slices.Contains(splitRoles(clf.Roles), defaults.RoleApp) {
+		if clf.AppName == "" {
+			return trace.BadParameter("to join application proxy to the cluster application name is required, provide it with --app-name or --name")
+		}
+
+		if clf.AppURI == "" && clf.AppCloud == "" {
```

Comment:
what if they set both?

Reply:
This is fine. For GCP and Azure this wouldn't change anything - right now we will only have CLI access for those - but for AWS setting URI is a requirement to gain AWS console access.